pkg/internal/controller/metrics/metrics.go

```diff
@@ -48,4 +48,10 @@ func init() {
 	metrics.Registry.MustRegister(QueueLength)
 	metrics.Registry.MustRegister(ReconcileErrors)
 	metrics.Registry.MustRegister(ReconcileTime)
+	metrics.Registry.MustRegister(
+		// expose process metrics like CPU, Memory, file descriptor usage etc.
+		prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}),
```

Comment:
I think it might be worth adding the a namespace to the `ProcessCollerctorOpts{}`, if we add the namespace `controller_runtime` then all metrics here will be prefixed with `controller_runtime_` and will fit in with the other metrics we are emitting from the library, WDYT?

```suggestion
		prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{Namespace: "controller_runtime"}),
```

Reply:
I thought about it and was a bit hesitant because these are not *really* controller-runtime specific metrics. If users are running other Go services (and k8s metrics for other components), they might already have configuration for processing these default prometheus metrics in their dashboards/pipelines. So keeping the same name might help them. WDYT ?

Reply:
Yeah good point, I've double checked with what the Kubernetes API server does and it doesn't prefix so let's keep it as is